### PR TITLE
Account for inherited types in new object detection.

### PR DESCRIPTION
**Pull request by DeathByTape**

### CODE REVIEW

File: elide-core/src/test/java/com/yahoo/elide/core/RequestScopeTest.java [hunk 1]
```diff
@@ -30,4 +35,39 @@ public void testFilterQueryParams() throws Exception {
...
+
+        String myId = "myId";
+        // Test that a new inherited class is counted for base type
+        requestScope.setUUIDForObject(dictionary.getJsonAliasFor(MyInheritedClass.class), myId, new MyInheritedClass());
+        Assert.assertNotNull(requestScope.getObjectById(dictionary.getJsonAliasFor(MyBaseClass.class), myId));
+    }
+
+    @Entity
+    @Include
+    public static class MyBaseClass {
```
Comment (aklish) [inline elide-core/src/test/java/com/yahoo/elide/core/RequestScopeTest.java]: Move this inside the test function.

File: elide-core/src/test/java/com/yahoo/elide/core/RequestScopeTest.java [hunk 2]
```diff
@@ -30,4 +35,39 @@ public void testFilterQueryParams() throws Exception {
...
+        @Id
+        public long id;
+    }
+
+    @Entity
+    @Include
+    public static class MyInheritedClass extends MyBaseClass {
```
Comment (aklish) [inline elide-core/src/test/java/com/yahoo/elide/core/RequestScopeTest.java]: Same

File: elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java [hunk 3]
```diff
@@ -379,4 +382,16 @@ private void bindTriggerIfPresent(Class<? extends Annotation> annotationClass, A
         }
         return annotation == NO_ANNOTATION ? null : (A) annotation;
     }
+
+    public static List<Class<?>> getInheritedTypes(Class<?> entityClass) {
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java]: Is there any reason this is here as opposed to the dictionary?

File: elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java [hunk 4]
```diff
@@ -410,10 +411,28 @@ public String getUUIDFor(Object o) {
...
     public void setUUIDForObject(String type, String id, Object object) {
         objectEntityCache.put(type, id, object);
+
+        // Insert for all inherited entities as well
+        dictionary.getSuperClassEntityNames(type).stream()
+                .map(i -> getInheritanceKey(type, i))
+                .forEach((newType) -> objectEntityCache.put(newType, id, object));
+    }
+
+    private String getInheritanceKey(String rootType, String superClass) {
```
Comment (aklish) [inline elide-core/src/main/java/com/yahoo/elide/core/RequestScope.java]: Should rootType be called subClass?

File: elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java [hunk 5]
```diff
@@ -379,4 +382,16 @@ private void bindTriggerIfPresent(Class<? extends Annotation> annotationClass, A
...
+        Class<?> cls = entityClass.getSuperclass();
+        ArrayList<Class<?>> results = new ArrayList<>();
+
+        while (cls != null && cls != Object.class) {
```
Comment (wcekan) [inline elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java]: Would this look any cleaner as
```java
for (Class<?> cls = entityClass.getSuperclass(); cls != null && cls != Object.class; cls = cls.getSuperclass()) {
    results.add(cls);
}
```

**Pull request merged**